server/methods.js

```diff
@@ -1,91 +1,174 @@
+const getChatterId = function(userId) {
+  const results = Chatter.User.find({userId: userId}).fetch();
+  if (results.length === 0) {
+    throw new Error("unknown-user", "user has not been added to chatter")
+  }
+  return results[0]._id;
+};
+
+const userInRoom = function(chatterId, roomId) {
+  const userRooms = Chatter.UserRoom.find({userId: chatterId, roomId}).fetch();
+  return userRooms.length > 0;
+};
+
+
 Meteor.methods({
-    "message.build" (params) {
-        return new Chatter.Message({
-            message: params.message,
-            roomId: params.roomId,
-            userId: params.userId,
-            userNick: getNickname(Meteor.user()),
-            userAvatar: getAvatar(Meteor.user())
-        }).save();
-    },
 
-    "userroom.build" (roomName) {
-        const userId = Meteor.userId();
-        const roomId = Chatter.Room.findOne({"name": roomName})._id;
-        const records = Chatter.UserRoom.find({"userId": userId, "roomId" : roomId}).fetch();
-
-        if (records === undefined || records.length === 0) {
-            return new Chatter.UserRoom({
-                roomId: roomId,
-                userId: userId
-            }).save();
-        }
-    },
+  "user.check" () {
+    const chatterUsers = Chatter.User.find({userId: Meteor.userId()}).fetch();
+    return chatterUsers.length > 0;
+  },
 
-    "room.build" (form) {
-        const user = Meteor.user();
-        return new Chatter.Room({
-            name: form.name,
-            roomType: form.roomType
-        }).save();
-    },
+  "message.build" (params) {
```

Comment:
I will changes all names and make them more intuitive. To be honest, I was using this notation since it was the style that was followed in the chatter boilerplate. I thought it was convention to use modelName.action, but I guess I can find  better ways of making this clearer